tools/build.py

```diff
@@ -35,149 +35,134 @@
 from tools.build_api import static_analysis_scan, static_analysis_scan_lib, static_analysis_scan_library
 from tools.build_api import print_build_results
 from tools.settings import CPPCHECK_CMD, CPPCHECK_MSG_FORMAT
+from utils import argparse_filestring_type
 
 if __name__ == '__main__':
     start = time()
 
     # Parse Options
     parser = get_default_options_parser()
 
-    parser.add_option("--source", dest="source_dir",
-                      default=None, help="The source (input) directory", action="append")
+    parser.add_argument("--source", dest="source_dir", type=argparse_filestring_type,
+                        default=None, help="The source (input) directory", action="append")
 
-    parser.add_option("--build", dest="build_dir",
+    parser.add_argument("--build", dest="build_dir", type=argparse_filestring_type,
```

Comment:
This sucker shouldn't have the `argparse_filestring_type` (since the build directory might not exist yet).